### PR TITLE
Driver: Controller latest update, KelvinVR Library for Prototype 0.5

### PR DESCRIPTION
**Pull request by NaturalAutomatics**
**KelvinVR** Library for MetaTemp project added, beta version for prototype 0.5

### CODE REVIEW

File: MetaTempDriver/Hardware/library/KelvinVR/examples/ledTest/KelvinVR_ledTest.ino [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+
+void setup() {
+  // put your setup code here, to run once:
+  fullPinSetup();
+  //blinkFrontLed(greenLed);
+}
+
+void loop() {
```
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/examples/ledTest/KelvinVR_ledTest.ino]: [nit] Remove all unused code later or move to another methods

File: MetaTempDriver/Hardware/ledTest/KelvinVR_ledTest/pinout.ino
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/ledTest/KelvinVR_ledTest/pinout.ino:1]: Create data structure to put all the constants in it as single item

File: MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+void Kelvin:: fadeEffect(int led) {
+   // set the brightness of pin 9:
+  analogWrite(led, brightness);
+
+  // change the brightness for next time through the loop:
+  _brightness = _brightness + _fadeAmount;
+
+  // reverse the direction of the fading at the ends of the fade:
+  if (_brightness <= 0 || _brightness >= 255) {
```
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.cpp]: define constants for each number here

File: MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  Main class for KelvinVR module
+*/
+
+#if defined(__AVR__)
+	#include "Arduino.h"
+	#include "hardware/avr/HW_AVR_defines.h"
+#elif defined(__PIC32MX__)
+	#include "WProgram.h"
+	#include "hardware/pic32/HW_PIC32_defines.h"
+#elif defined(__arm__)
```
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.h]: [nit] Create support for ESP32 Platform in the future

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM.h
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM.h:1]: DELETE THIS OLED Library preset file!

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM_defines.h
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM_defines.h:1]: DELETE this from preset

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR.h
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR.h:1]: DELETE this preset from OLED library

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR_defines.h
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR_defines.h:1]: DELETE this from OLED preset

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/pic32/HW_PIC32.h
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/hardware/pic32/HW_PIC32.h:1]: DELETE this file from OLED preset

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/pic32/HW_PIC32_defines.h
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/KelvinVR/hardware/pic32/HW_PIC32_defines.h:1]: DELETE this file from OLED preset

File: MetaTempDriver/Hardware/library/library.ino
Comment (NaturalAutomatics) [inline MetaTempDriver/Hardware/library/library.ino:1]: What's that for?

**Review by NaturalAutomatics — COMMENTED** [review]
Please remove all unnecessary files here

File: MetaTempDriver/Hardware/library/KelvinVR/examples/ledTest/KelvinVR_ledTest.ino (hunk 1 above)
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/examples/ledTest/KelvinVR_ledTest.ino]: Sure, created task for it (ticket https://naturalautomatics.atlassian.net/browse/MTP-12)

File: MetaTempDriver/Hardware/ledTest/KelvinVR_ledTest/pinout.ino
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/ledTest/KelvinVR_ledTest/pinout.ino:1]: It suppose to be [nit] here, refactoring will be in ticket (https://naturalautomatics.atlassian.net/browse/MTP-12)

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM_defines.h
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM_defines.h:1]: Done, deleted

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR.h
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR.h:1]: Deleted

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR_defines.h
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/hardware/avr/HW_AVR_defines.h:1]: +

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/pic32/HW_PIC32.h
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/hardware/pic32/HW_PIC32.h:1]: Done!

**Comment by AlexNoyanov** [comment]
> Please remove all unnecessary files here

Deleted all files from all commits here, updated platform support

File: MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.cpp (hunk 2 above)
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.cpp]: Done!

File: MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.h (hunk 3 above)
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/KelvinVR.h]: Created new files for ESP32 and Arduino

File: MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM.h
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/KelvinVR/hardware/arm/HW_ARM.h:1]: +

File: MetaTempDriver/Hardware/library/library.ino
Comment (AlexNoyanov) [reply MetaTempDriver/Hardware/library/library.ino:1]: Deleted!

**Pull request merged**